### PR TITLE
Update castling rules

### PR DESCRIPTION
**Pull request by losos0**

### CODE REVIEW

**Review by dstrain115 — COMMENTED** [review]
Do you think you could explain what the differences are (in the PR description)?  This would help to review it.

**Comment by losos0** [comment]
The existing implementation always measured the target squares (if potentially occupied), while there is supposed to be an EXCLUDED variant that measures and a BASIC variant that doesn't. The BASIC variant is needed because you are allowed to castle without measuring if there is a same-color rook in superposition on the rook's target square.

File: unitary/quantum_chess/quantum_board.py [hunk 1]
```diff
@@ -660,24 +665,72 @@ def _clear_path_ancilla(self, path_qubits, ancilla):
...
+                self.set_castle(sbit, rook_sbit, tbit, rook_tbit)
+                return 1
+
+            self.state = set_nth_bit(sbit, self.state, False)
+            self.state = set_nth_bit(tbit, self.state, True)
+            self.add_entangled(rook_squbit, rook_tqubit)
+            self.circuit.append(qm.normal_move(rook_squbit, rook_tqubit))
+            return 1
+
+        elif variant == enums.MoveVariant.EXCLUDED:
```
Comment (dstrain115) [inline unitary/quantum_chess/quantum_board.py:708]: Maybe we could add some comments here explaining the two versions of the castle as you did in the PR?

**Review by dstrain115 — APPROVED** [review]

File: unitary/quantum_chess/quantum_board.py (hunk 1 above)
Comment (losos0) [reply unitary/quantum_chess/quantum_board.py:712]: Done.

**Pull request merged**